RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalMultiCluster.cc

```diff
@@ -0,0 +1,67 @@
+#include "PFClusterFromHGCalMultiCluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/Event.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+void PFClusterFromHGCalMultiCluster::updateEvent(const edm::Event& ev) {
+  ev.getByToken(_clusterToken, _clusterH);
+}
+
+void PFClusterFromHGCalMultiCluster::buildClusters(
+    const edm::Handle<reco::PFRecHitCollection>& input,
+    const std::vector<bool>& rechitMask, const std::vector<bool>& seedable,
+    reco::PFClusterCollection& output) {
+  // USE HGCal Multiclusters instead
+
+  const auto& hgcalMultiClusters = *_clusterH;
+  auto const& hits = *input;
+
+  // for quick indexing back to hit energy
+  std::unordered_map<uint32_t, size_t> detIdToIndex(hits.size());
+  for (uint32_t i = 0; i < hits.size(); ++i) {
+    detIdToIndex[hits[i].detId()] = i;
+  }
+
+  for (const auto& mcl : hgcalMultiClusters) {
+    edm::Ref<std::vector<reco::PFRecHit> > seed;
+    double energy = 0.0, highest_energy = 0.0;
+    output.emplace_back();
+    reco::PFCluster& back = output.back();
+    for (const auto& cl : mcl) {
+      const auto& hitsAndFractions = cl->hitsAndFractions();
+      for (const auto& hAndF : hitsAndFractions) {
+        auto itr = detIdToIndex.find(hAndF.first);
+        if (itr == detIdToIndex.end()) {
+	  continue;  // hit wasn't saved in reco
+	}
+        auto ref = makeRefhit(input, itr->second);
+	assert(ref->detId() == hAndF.first.rawId());
+        const double hit_energy = hAndF.second * ref->energy();
+        energy += hit_energy;
+        back.addRecHitFraction(reco::PFRecHitFraction(ref, hAndF.second));
+        // TODO: the following logic to identify the seed of a cluster
+        // could be appropriate for the Run2 Ecal Calorimetric
+        // detector, but could be wrong for the HGCal one. This has to
+        // be reviewd.
+        if (hit_energy > highest_energy || highest_energy == 0.0) {
+          highest_energy = hit_energy;
+          seed = ref;
+        }
+      }  // end of hitsAndFractions
+    }    // end of loop over clusters (2D/layer)
+    if (energy <= 1) {
```

Comment:
please add a comment to clarify the choice of 1.0 is reasonable, perhaps in some signal and noise terms {why not 0.7 or 3 or 0.02 as a random other set of values}